sdx_controller/handlers/connection_handler.py

```diff
@@ -366,13 +366,13 @@ def handle_link_failure(self, te_manager, failed_links):
                     logger.info(
                         f"Connection {service_id} affected by link {link['id']}"
                     )
-                    connection_str = self.db_instance.read_from_db(
+                    connection = self.db_instance.read_from_db(
                         MongoCollections.CONNECTIONS, service_id
                     )
-                    if not connection_str:
+                    if not connection:
                         logger.debug(f"Did not find connection from db: {service_id}")
                         continue
-                    connection = json.loads(connection_str[service_id])
+                    connection = connection[service_id]
```

Comment:
here again we could benefit from having this dict key access encapsulated on the `read_from_db()`